connector/oidc/oidc.go

```diff
@@ -190,41 +195,49 @@ func (c *oidcConnector) HandleCallback(s connector.Scopes, r *http.Request) (ide
 		return identity, fmt.Errorf("oidc: failed to verify ID Token: %v", err)
 	}
 
-	var claims struct {
-		Username      string `json:"name"`
-		Email         string `json:"email"`
-		EmailVerified bool   `json:"email_verified"`
-		HostedDomain  string `json:"hd"`
-	}
+	var claims map[string]interface{}
 	if err := idToken.Claims(&claims); err != nil {
 		return identity, fmt.Errorf("oidc: failed to decode claims: %v", err)
 	}
 
+	name := claims["name"].(string)
```

Comment:
❓ Does this have the same error semantics as the struct before? I.e. should the `idToken` _not_ have an `email` key, this would explode -- `claims["email"]` would be nil, `.(string)` would fail. What used to happen with `idToken.Claims(&claims)`? (Genuine question, I haven't checked...)

Reply:
Thanks for your feedback!
I fixed it:+1: I added error handlings for missing specified claims.
https://github.com/dexidp/dex/pull/1448/commits/601fa713aa8617640cea21ed92052c023d5ca72d